stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt

```diff
@@ -23,30 +23,40 @@ import android.net.NetworkRequest
 import android.os.Build
 import io.getstream.log.taggedLogger
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
 import java.util.concurrent.atomic.AtomicBoolean
 
 internal class NetworkStateProvider(
     private val scope: CoroutineScope,
     private val connectivityManager: ConnectivityManager,
 ) {
-
     private val logger by taggedLogger("Chat:NetworkStateProvider")
     private val lock: Any = Any()
     private val callback = object : ConnectivityManager.NetworkCallback() {
         override fun onAvailable(network: Network) {
-            notifyListenersIfNetworkStateChanged()
+            // Note: It is not safe to call getNetworkCapabilities(Network) here, as the ConnectivityManager state might
+            // not be updated yet (see NetworkCallback#onAvailable(Network) documentation).
+            // Therefore, we introduce a delay before notifying listeners, to ensure the ConnectivityManager is updated.
+            notifyListenersIfNetworkStateChangedAsync()
         }
 
         override fun onCapabilitiesChanged(network: Network, networkCapabilities: NetworkCapabilities) {
-            notifyListenersIfNetworkStateChanged()
+            notifyListenersIfNetworkStateChangedAsync()
         }
 
         override fun onLost(network: Network) {
-            notifyListenersIfNetworkStateChanged()
+            // Note: It is not safe to call getNetworkCapabilities(Network) here, as the ConnectivityManager state might
+            // not be updated yet (see NetworkCallback#onLost(Network) documentation).
+            // Therefore, we introduce a delay before notifying listeners, to ensure the ConnectivityManager is updated.
+            notifyListenersIfNetworkStateChangedAsync()
```

Comment:
Instead of introducing a delay, why don't we directly notify our listener the connection was lost? :thinking: